database/src/main/resources/migrations/ddl/V1.0.1__notifications_migration.sql

```diff
@@ -0,0 +1,22 @@
+CREATE TABLE notifications
+(
+    id             INT GENERATED ALWAYS AS IDENTITY,
+    user_id        INT       NOT NULL,
+    email          VARCHAR(255),
+    marked_as_read BOOLEAN DEFAULT FALSE,
+    created_at     TIMESTAMP NOT NULL,
+    payload        VARCHAR(255) NOT NULL,
+    type           VARCHAR(255),
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE notifications_channels
+(
+    id              INT GENERATED ALWAYS AS IDENTITY ,
```

Comment:
Don't think we'll need the ID
The channel itself won't be a self-standing entity.
Instead we can use an unique index on the column pair (notification_id, channel)